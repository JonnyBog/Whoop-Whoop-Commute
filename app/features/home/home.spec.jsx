import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';

import Home from 'features/home/home';

describe('Features', () => {
  describe('Home component', () => {
    it('should export a function', () => {
      expect(Home).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        data:{
          currentForecast: [
            {
              forecastSummary: 'test'
            }
          ]
        }
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createShallowRenderTree
     * @returns {JSX} shallow renderer component
     */
    function createShallowRenderTree () {
      const shallowRenderer = rendererShallow.createRenderer();

      return shallowRenderer.render(
        <Home {...props} />
      );
    }

    it('should return content', () => {
      const tree = createShallowRenderTree();
      expect(tree).toMatchSnapshot();
    });
  });
});
