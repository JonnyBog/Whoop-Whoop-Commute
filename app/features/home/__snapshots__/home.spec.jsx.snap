// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Home component should return content 1`] = `
<home__HomeContainer>
  <GridItem
    py={
      Array [
        "6px",
        "8px",
        "12px",
      ]
    }
    width={
      Array [
        1,
        1,
        1,
      ]
    }
  >
    <div>
      test
    </div>
  </GridItem>
</home__HomeContainer>
`;
